docs/sources/lids/_DRAFT_RemoteRuleEvaluation.md

```diff
@@ -0,0 +1,84 @@
+
+Recording rules produce metric series which are sent to a Prometheus-compatible source. Alerting rules send notifications to Alertmanager when a condition is met. Both of these rule types can play a vital role in an organisation's observability strategy, and so their reliable evaluation is essential.
+
+## Problem Statement
+
+Rule evaluations can contain expensive queries. The `ruler` initialises a `querier`, but the `querier` does not have the capability to accelerate queries; the `query-frontend` component is responsible for query acceleration through splitting, sharding, caching, and other techniques.
+
+An expensive rule query can cause an entire `ruler` instance to use excessive resources and even crash. This is highly problematic for the following reasons:
+
+- slow rule evaluations can lead to subsequent rules in a group to be delayed or missed, leading to missing alerts or gaps in recording rule metrics
+- excessive resource usage can impede the evaluation of rules for other tenants (noisy neighbour)
+
+## Goals
+
+- faster, more efficient rule evaluation
+- greater isolation between tenants
+- more reliable service
+
+## Non-Goals
+
+This proposal does not aim to make this option the default mode of evaluation; it should be optional because it increases operational complexity.
+
+## Proposals
+
+### Proposal 0: Do nothing
+
+Loki's current `ruler` implementation is sufficient for small installations running relatively simple or inexpensive queries.
+
+**Pros**:
+- Nothing to be done
+
+**Cons:**
+- Loki's `ruler` will remain unreliable and inefficient when used in large multi-tenant environments with expensive queries.
+
+### Proposal 1: Remote Execution
+
+Taking inspiration from [Grafana Mimir's implementation](https://grafana.com/docs/mimir/latest/operators-guide/architecture/components/ruler/#remote), the `ruler` would be configured to send its rule query to the `query-frontend` component over gRPC. The `querier` instances receiving queries from the `query-frontend` (or optionally via the `query-scheduler`) will handle the request and send the responses to the `query-frontend` and be combined. The `ruler` will receive and process these responses as if the query had been executed locally.
+
+**Pros:**
+- Takes full advantage of Loki's query acceleration techniques, leading to faster and more efficient rule evaluation
+- Operationally simple as existing `query-frontend`/`query-scheduler`/`querier` setup can be used
+- Per-tenant isolation available in Loki's query path (shuffle-sharding, per-tenant queues) can be used to reduce or eliminate the noisy neighbour problem
+
+**Cons:**
+- Increased interdependence in components, increased cross-component networking
+- Reusing the same `query-frontend`/`query-scheduler`/`querier` setup can cause expensive queries to starve rule evaluations of query resources, and vice versa
+  - Additional complexity introduced if this setup needs to be duplicated for rule evaluations (recommended)
```

Comment:
Since you say that separating the rule evaluation path from the regular query path is the recommended way, I think it makes sense to add a few words about that preferred setup. While you describe the problem with a common query and rule execution path in the section "Other Notes" below, you don't describe the possible solution.

Reply:
> this setup needs to be duplicated

I think that's describing what needs to be done, no? Maybe I should make this a bit more clear.

Reply:
Yes, kind of, but I think it should be more prominent as part of the proposal if it's the recommended way to run the rule execution in the future.
I guess it's a similar concern that @salvacorts had in [his comment](https://github.com/grafana/loki/pull/8129/files#r1069481832). My comment should not block the LID from being merged, though.